packages/wallet/test/hardware/ledger/LedgerKeyAgent.test.ts

```diff
@@ -298,6 +306,258 @@ describe('LedgerKeyAgent', () => {
+              )
+            }
+          ];
+          const tx = txBuilder
+            .addOutput(
+              txBuilder.buildOutput().address(outputs[0].address).coin(BigInt(outputs[0].value.coins)).toTxOut()
+            )
+            .build();
+
+          await expect(tx.sign()).rejects.toThrow(
+            InvalidDataReason.SIGN_MODE_ORDINARY__CERTIFICATE_STAKE_CREDENTIAL_ONLY_AS_PATH
+          );
+        });
+
+        it('can sign a transaction with Unregistration certs', async () => {
+          const txBuilder = wallet.createTxBuilder();
+          txBuilder.partialTxBody.certificates = [
+            {
+              __typename: Cardano.CertificateType.Unregistration,
+              deposit: 5n,
+              stakeCredential: getStakeCredential(
+                (await firstValueFrom(wallet.delegation.rewardAccounts$))?.[0].address
+              )
+            }
+          ];
+          const tx = txBuilder
+            .addOutput(
+              txBuilder.buildOutput().address(outputs[0].address).coin(BigInt(outputs[0].value.coins)).toTxOut()
+            )
+            .build();
+
+          expect(await tx.sign()).toBeTruthy();
+        });
+
+        it('would throw while trying to sign a transaction with Unregistration certs', async () => {
+          const txBuilder = wallet.createTxBuilder();
+          txBuilder.partialTxBody.certificates = [
+            {
+              __typename: Cardano.CertificateType.Unregistration,
+              deposit: 5n,
+              stakeCredential: getStakeCredential(
+                Cardano.RewardAccount('stake_test1up7pvfq8zn4quy45r2g572290p9vf99mr9tn7r9xrgy2l2qdsf58d')
+              )
+            }
+          ];
+          const tx = txBuilder
+            .addOutput(
+              txBuilder.buildOutput().address(outputs[0].address).coin(BigInt(outputs[0].value.coins)).toTxOut()
+            )
+            .build();
+
+          await expect(tx.sign()).rejects.toThrow(
+            InvalidDataReason.SIGN_MODE_ORDINARY__CERTIFICATE_STAKE_CREDENTIAL_ONLY_AS_PATH
+          );
+        });
+
+        it('can sign a transaction with VoteDelegation certs', async () => {
+          const credential = getStakeCredential((await firstValueFrom(wallet.delegation.rewardAccounts$))?.[0].address);
+          const txBuilder = wallet.createTxBuilder();
+          txBuilder.partialTxBody.certificates = [
+            {
+              __typename: Cardano.CertificateType.VoteDelegation,
+              dRep: credential,
+              stakeCredential: credential
+            }
+          ];
+          const tx = txBuilder
+            .addOutput(
+              txBuilder.buildOutput().address(outputs[0].address).coin(BigInt(outputs[0].value.coins)).toTxOut()
+            )
+            .build();
+
+          // expect(await tx.sign()).toBeTruthy();
+          // TODO: should be able to sign?
+          await expect(tx.sign()).rejects.toThrow(
```

Comment:
@AngelCastilloB could you please suggest what might be an issue here?

Reply:
signing the tx with VoteDelegation cert throws `HwMappingError: Hardware data mapping failure: Ledger computed a different transaction id`, i've probably mocked it in a wrong way?

Reply:
The ledger device reconstructs internally the transaction and hashes it, if the hash is not matching, it probable means we are not encoding the transaction in canonical form on our side, or we are not mapping all the data the device needs to compute the hashes, make sure the object that is being finally passed to the ledger device contains all the data of the transaction, and that all arrays are canonically sorted

Reply:
resolved, thanx @AngelCastilloB